R/data.table.R

```diff
@@ -147,6 +147,33 @@ replace_dot_alias = function(e) {
     on.exit(options(oldverbose))
   }
   .global$print=""
+  # with2 #4485
+  if (!missing(with) && (length(with)>1L || !is.null(names(with)))) {
+    w = wither(with)
+    # this checks for valid negation usage in 'j' should be removed after some time, it is here to ensure newcomers to this feature will use it correctly
+    if (isFALSE(w[["j"]]) && !identical(as.integer(getOption("datatable.with",1L)), 2L) && hasNot(subj<-substitute(j)) &&
+        !with_j_valid(subj, env=parent.frame(), verbose=verbose)) stop("internal error: with_j_valid should have raised an error already") # nocov #w$j = NA
+    nr = nrow(x)
+    nc = length(x)
+    if (isFALSE(w[["i"]]) && !missing(i)) i = with_i(i, len=nr, verbose=verbose)
+    if (isFALSE(w[["j"]]) && !missing(j)) j = with_j(j, len=nc, x=x, verbose=verbose)
+    if ((isFALSE(w[["i"]]) && missing(j)) || (isFALSE(w[["j"]]) && missing(i)) || (isFALSE(w[["i"]]) && isFALSE(w[["j"]]))) {
```

Comment:
a bit hard to figure out what this branch is for

Reply:
from body of this branch one can see that branch is for early return and escape rest of [ processing